### PR TITLE
Ticket-8

### PR DESCRIPTION
**Pull request by yuridevweb**

### CODE REVIEW

**Review by giteron — CHANGES_REQUESTED** [review]
Nice inventive way of adding comment counts! However, this could be done with a single database query. This would help the efficiency of your getArticles function, especially if it had a large data set to work with (e.g. 100s-1000s of articles).

You have most of the syntax for generating a comment count, so it should just be a case of combining the two requests :)

File: __tests__/app.test.js [hunk 1]
```diff
@@ -171,15 +171,31 @@ describe('GET /api/articles', () => {
         articles.forEach((article) => {
           expect(article).toEqual(
             expect.objectContaining({
+              article_id: expect.any(Number),
               author: expect.any(String),
               body: expect.any(String),
```
Comment (jadelandeg) [inline __tests__/app.test.js]: check the trello for what we expect in this endpoint, don't want to see the body when getting all articles.

File: controllers/news.controllers.js [hunk 2]
```diff
@@ -41,15 +45,35 @@ exports.patchArticleById = (req, res, next) => {
...
-    console.log(articles, '<<<<<<<<<')
-    res.status(200).send({ articles })
+    const promises = []
+    articles.forEach((article) => {
+      promises.push(commentsByArticle(article.article_id))
+    })
+    Promise.all(promises)
+      .then((results) => {
+        let articleWithComments = articles.map((article, index) => {
+          article.comment_count = Number(results[index].count)
```
Comment (jadelandeg) [inline controllers/news.controllers.js:65]: should be doing get articles with comment count in one query here, not using a promise.all. Shouldn't use the length, should take advantage of SQL functions.

File: models/news.models.js [hunk 3]
```diff
@@ -40,8 +40,22 @@ exports.selectUsers = () => {
...
 exports.selectArticles = () => {
-  return db.query('SELECT * FROM articles').then((result) => {
+  let queryStr = `SELECT articles.*, 
+    COUNT(comments.comment_id) AS comment_count FROM articles
+    LEFT JOIN comments
+    ON comments.article_id = articles.article_id
+    GROUP BY articles.article_id
+    ORDER BY created_at DESC;`
+
+  return db.query(queryStr).then((result) => {
```
Comment (jadelandeg) [inline models/news.models.js:59]: looking great!

Comment (jadelandeg) [reply models/news.models.js:59]: just a minor change to stop the body coming back

**Review by jadelandeg — CHANGES_REQUESTED** [review]
Good job, Yuri! Just a little change this time. Make sure the query is returning the keys listed on the trello ticket :)

**Comment by cavalierswain** [comment]
Good effort on implementing the changes!
looks good!

**Pull request merged**